src/app/services/case/case.service.ts

```diff
@@ -0,0 +1,78 @@
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, throwError } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import { ErrorHandlerService } from '../error/error-handler.service';
+import { CaseData } from '../../../app/types/case';
+
+const GET_CASE_PATH = '/api/cases/';
+const ADVANCED_SEARCH_CASE_PATH = '/api/cases/search';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CaseService {
+  constructor(private readonly http: HttpClient, private ErrorHandler: ErrorHandlerService) {}
+
+  //Single get case API
+  getCase(caseId: string | number): Observable<CaseData> {
+    const apiURL = `${GET_CASE_PATH}?caseId=${caseId}`;
+    return this.http.get<CaseData>(apiURL).pipe(
+      catchError((err: Error) => {
+        this.ErrorHandler.handleError(err);
+        return throwError(() => err);
+      })
+    );
+  }
```

Comment:
Is this one used yet? The get case by ID endpoint uses a path param ie. `/cases/:caseId` rather than query param.

Reply:
Not used yet but will make sure it's updated when the time comes